src/NServiceBus.Core/Pipeline/Incoming/ReceiveFeature.cs

```diff
@@ -65,6 +66,7 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction
             internal static readonly Task<CompletableSynchronizedStorageSession> EmptyResult = Task.FromResult<CompletableSynchronizedStorageSession>(new NoOpCompletableSynchronizedStorageSession());
         }
 
+        [SkipWeaving]
```

Comment:
you may want to add a comment why we don't want janitor to implement disposable for this class.